src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -186,15 +186,26 @@ PaymentSandbox::balanceHook (AccountID const& account,
     {
         auto delta = amount.zeroed ();
         auto lastBal = amount;
+        auto minBal = amount;
         for (auto curSB = this; curSB; curSB = curSB->ps_)
         {
             if (auto adj = curSB->tab_.adjustments (account, issuer, currency))
             {
                 delta += adj->debits;
                 lastBal = adj->origBalance;
```

Comment:
The `origBalance` is the balance from the ledger before any debits/credits to that trust line _on that payment sandbox_. So `orgBalance` in face will be different on each level on the stack of sandboxes.

You ask how it pays into the code. We take the minimum of these values as a lower bound for our return value. The reason this is needed is:

1) Sometimes the original balance - all the debits is slightly larger than it should be due to loss of precision in the accumulated debits.
2) We already use the current balance as a lower bound, but the account may have received a credit, so that lower bound is too large.

With this scheme, when a sandbox is thrown away, the available balance will never go down (as required by the payment algorithm).